org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundGenerator.java

```diff
@@ -157,6 +157,7 @@ public String getDescription() {
 		
 		List<IPosition> points = new ArrayList<>(size());
 		createPoints(0, points, null);
+		for (int i = 0; i < points.size(); i++) points.get(i).setStepIndex(i);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)